lambda/repository/lambda_functions.py

```diff
@@ -54,13 +55,130 @@ def _get_embeddings(model_name: str, id_token: str) -> LisaOpenAIEmbeddings:
         lisa_api_endpoint = lisa_api_param_response["Parameter"]["Value"]
 
     base_url = f"{lisa_api_endpoint}/{os.environ['REST_API_VERSION']}/serve"
+    cert_path = get_cert_path(iam_client)
 
     embedding = LisaOpenAIEmbeddings(
-        lisa_openai_api_base=base_url, model=model_name, api_token=id_token, verify=get_cert_path(iam_client)
+        lisa_openai_api_base=base_url, model=model_name, api_token=id_token, verify=cert_path
     )
     return embedding
 
 
+def _get_embeddings_pipeline(model_name: str) -> Any:
+    """
+    Get embeddings for pipeline requests using management token.
+
+    Args:
+        model_name: Name of the embedding model to use
+
+    Raises:
+        ValidationError: If model name is invalid
+        Exception: If API request fails
+    """
+    logger.info("Starting pipeline embeddings request")
+    validate_model_name(model_name)
+
+    # Create embeddings client that matches LisaOpenAIEmbeddings interface
+    class PipelineEmbeddings:
```

Comment:
Can you pull this class up a level to be outside the function? You'll need to pass in the model_name as part of the constructor. Otherwise, the cyclomatic complexity is pretty high (22).